components/clarity-repl/src/analysis/check_checker/mod.rs

```diff
@@ -3023,6 +3025,31 @@ mod tests {
         };
     }
 
+    #[test]
+    fn allow_unchecked_bool_in_public_function() {
+        let mut settings = SessionSettings::default();
+        settings.repl_settings.analysis.passes = vec![Pass::CheckChecker];
+        let mut session = Session::new(settings);
+        let snippet = "
+(define-public (allow_tainted (amount bool))
+    (ok amount)
+)
```

Comment:
This contract would not trigger a check-checker warning now. Returning tainted data from a public function is allowed. How about using the tainted data to set a data var instead:

```clarity
(define-data-var myvar bool false)
(define-public (tainted-var-set (b bool))
    (ok (var-set myvar b))
)
```

Reply:
Ah didn't know that, I updated to use your example.. and learned something 😅